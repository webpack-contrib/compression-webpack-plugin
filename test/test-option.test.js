import CompressionPlugin from '../src/index';

import {
  compile,
  getAssetsNameAndSize,
  getCompiler,
  getErrors,
  getWarnings,
  removeCache,
} from './helpers/index';

describe('"test" option', () => {
  let compiler;

  beforeEach(() => {
    compiler = getCompiler(
      './entry.js',
      {},
      {
        output: {
          path: `${__dirname}/dist`,
          filename: '[name].js?var=[hash]',
          chunkFilename: '[id].[name].js?ver=[hash]',
        },
      }
    );

    return removeCache();
  });

  it('matches snapshot with empty `test` value', async () => {
    new CompressionPlugin({
      minRatio: 1,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(getAssetsNameAndSize(stats)).toMatchSnapshot('assets');
    expect(getWarnings(stats)).toMatchSnapshot('errors');
    expect(getErrors(stats)).toMatchSnapshot('warnings');
  });

  it('matches snapshot for a single `test` value ({RegExp})', async () => {
    new CompressionPlugin({
      test: /\.(png|jpg|gif)$/i,
      minRatio: 1,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(getAssetsNameAndSize(stats)).toMatchSnapshot('assets');
    expect(getWarnings(stats)).toMatchSnapshot('errors');
    expect(getErrors(stats)).toMatchSnapshot('warnings');
  });

  it('matches snapshot for multiple `test` values ({Array<RegExp>})', async () => {
    new CompressionPlugin({
      test: [/\.(png|jpg|gif)$/i, /\.svg/i],
      minRatio: 1,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(getAssetsNameAndSize(stats)).toMatchSnapshot('assets');
    expect(getWarnings(stats)).toMatchSnapshot('errors');
    expect(getErrors(stats)).toMatchSnapshot('warnings');
  });
});
