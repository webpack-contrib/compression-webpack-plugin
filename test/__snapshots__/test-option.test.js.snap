// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with \`test\` option matches snapshot for a single \`test\` value: assets 1`] = `
Array [
  "1.async.js?ver=1ea6f97e389bd951fd50",
  "5b1f36bc41ab31f5b801d48ba1d65781.png",
  "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
  "96621d3c37d96ad3bf792fcc848d912f.svg",
  "js.js?var=1ea6f97e389bd951fd50",
]
`;

exports[`when applied with \`test\` option matches snapshot for a single \`test\` value: errors 1`] = `Array []`;

exports[`when applied with \`test\` option matches snapshot for a single \`test\` value: warnings 1`] = `Array []`;

exports[`when applied with \`test\` option matches snapshot for multiple \`test\` values: assets 1`] = `
Array [
  "1.async.js?ver=1ea6f97e389bd951fd50",
  "5b1f36bc41ab31f5b801d48ba1d65781.png",
  "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
  "96621d3c37d96ad3bf792fcc848d912f.svg",
  "96621d3c37d96ad3bf792fcc848d912f.svg.gz",
  "js.js?var=1ea6f97e389bd951fd50",
]
`;

exports[`when applied with \`test\` option matches snapshot for multiple \`test\` values: errors 1`] = `Array []`;

exports[`when applied with \`test\` option matches snapshot for multiple \`test\` values: warnings 1`] = `Array []`;

exports[`when applied with \`test\` option matches snapshot with empty \`test\` value: assets 1`] = `
Array [
  "1.async.js.gz?ver=1ea6f97e389bd951fd50",
  "1.async.js?ver=1ea6f97e389bd951fd50",
  "5b1f36bc41ab31f5b801d48ba1d65781.png",
  "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
  "96621d3c37d96ad3bf792fcc848d912f.svg",
  "96621d3c37d96ad3bf792fcc848d912f.svg.gz",
  "js.js.gz?var=1ea6f97e389bd951fd50",
  "js.js?var=1ea6f97e389bd951fd50",
]
`;

exports[`when applied with \`test\` option matches snapshot with empty \`test\` value: errors 1`] = `Array []`;

exports[`when applied with \`test\` option matches snapshot with empty \`test\` value: warnings 1`] = `Array []`;
